hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/FSSchedulerNode.java

```diff
@@ -118,23 +178,56 @@ synchronized FSAppAttempt getReservedAppSchedulable() {
    *
    * @param containers container to mark
    */
-  void addContainersForPreemption(Collection<RMContainer> containers) {
-    containersForPreemption.addAll(containers);
+  void addContainersForPreemption(Collection<RMContainer> containers,
+                                  FSAppAttempt appAttempt) {
+    for(RMContainer container : containers) {
+      containersForPreemption.put(container, appAttempt);
```

Comment:
We should add the entry to the other map also here. That way, we count allocations prior to the preemptions also against the same app.